index.js

```diff
@@ -357,7 +358,7 @@ class ReadableState {
   updateNextTick () {
     if ((this.stream._duplexState & READ_NEXT_TICK) !== 0) return
     this.stream._duplexState |= READ_NEXT_TICK
-    process.nextTick(updateReadNT, this)
+    queueMicrotask(updateReadNT.bind(this, this))
```

Comment:
We should just bind this once in the constructor similar to how we do it with afterRead (and change updateReadNT to use `this` like afterRead does.

Reply:
elaborate?

Reply:
The "nextTicked" function do not need any arguments, so the bind here is always the same for the instance.
So we can save it in the constructor instead to avoid the function allocation every-time we queue it.

Like

``` js
constructor () {
  ...
  this._updateNT = updateReadNT.bind(this)
}
```

And then change updateReadNT to

``` js
function updateReadNT () {
  this.stream._duplexState &= READ_NOT_NEXT_TICK
  this.update()
}
```

Makes sense?

Reply:
(We passed down `this` before as an argument because nextTick supports that, but microqueue doesn't)

Reply:
ah, I see what I overlooked, my bad

Reply:
@mafintosh is this good?